pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/data/ParticipantRecord.java

```diff
@@ -0,0 +1,101 @@
+import org.broadinstitute.dsm.util.export.ParticipantExportPayload;
+
+@Slf4j
+@Data
+public class ParticipantRecord {
+    private String ddpParticipantId;
+    private int participantId;
+    private int participantRecordId;
+    private DDPInstanceDto ddpInstanceDto;
+    private String fieldTypeId;
+    private ParticipantExportPayload participantExportPayload;
+    private Participant participant;
+
+    private ParticipantRecordDao participantRecordDao;
+    private Map<String, String> data;
+
+
+    public ParticipantRecord(String ddpParticipantId, DDPInstanceDto ddpInstanceDto, ParticipantRecordDao participantRecordDao) {
+        this.ddpParticipantId = ddpParticipantId;
+        this.ddpInstanceDto = ddpInstanceDto;
+        this.participantRecordDao = participantRecordDao;
+    }
+
+    public int createNewParticipantRecord() {
+        participantId = insertDdpParticipant();
+        participantRecordId = insertDdpParticipantRecord(participantId);
+        return participantId;
+    }
+
+    private int insertDdpParticipant() {
+        ParticipantDto participantDto =
+                new ParticipantDto.Builder(ddpInstanceDto.getDdpInstanceId(), System.currentTimeMillis())
+                        .withDdpParticipantId(ddpParticipantId)
+                        .withLastVersion(0)
+                        .withLastVersionDate("")
+                        .withChangedBy("SYSTEM")
+                        .build();
+
+        int participantId = new ParticipantDao().create(participantDto);
+        participantExportPayload = new ParticipantExportPayload(
+                participantId,
+                ddpParticipantId,
+                String.valueOf(ddpInstanceDto.getDdpInstanceId()),
+                ddpInstanceDto.getInstanceName(),
+                ddpInstanceDto
+        );
+        ElasticSearchParticipantExporterFactory.fromPayload(
+                participantExportPayload
+        ).export();
+
+        return participantId;
+    }
+
+    private int insertDdpParticipantRecord(int participantId) {
+        ParticipantRecordDto participantRecordDto =
+                new ParticipantRecordDto.Builder(participantId, System.currentTimeMillis())
+                        .withChangedBy("SYSTEM")
+                        .build();
+        return participantRecordDao.create(participantRecordDto);
+    }
+
+    public boolean insertDefaultValues(Map<String, String> data, int participantId) {
+        String additionalValues = new Gson().toJson(data);
+        participantRecordDao.insertDefaultAdditionalValues(participantId, additionalValues);
+        participant = new DefaultParticipantExporter(participantExportPayload).buildParticipantFromPayload(participantExportPayload);
+        participant.setAdditionalValuesJson(additionalValues);
+        String participantGuid = Exportable.getParticipantGuid(ddpParticipantId, ddpInstanceDto.getEsParticipantIndex());
```

Comment:
I believe `ddpParticipantId` is the `guid` so we don't need to read it, or am I missing something?

Reply:
@pegahtah you already have `guid` in `SingularDefaultValues`, you may pass it to this class to avoid extra ES call here.

Reply:
Thanks folks ! I changed it!